harness/determined/experimental/core_v2/_core_v2.py

```diff
@@ -279,3 +296,44 @@ def url_reverse_webui_exp_view() -> str:
 
     assert _client is not None
     return core_v2._url_reverse_webui_exp_view(_client, exp_id)
+
+
+def _merge_config(
+    config: Optional[Config],
+    defaults: Optional[DefaultConfig],
+    unmanaged: Optional[UnmanagedConfig],
+) -> Config:
+    if defaults is not None or unmanaged is not None:
```

Comment:
nit: this method is a little convoluted with the combined if/elses around unmanaged vs which configs are passed in. partially my fault, i thought separating out the config merging logic would make it easier to read, but i see now we need all these checks in different places.

maybe if we just bubbled all these checks up to both of the callers (`init` and `init_context`) it'd be less confusing? i don't know, maybe like:
```python3
def init/init_context(...):
...
    # first check for deprecated fields and log warning
    if defaults is not None or unmanaged is not None:
        ...
    
    # now check for on cluster/trial and just return if we are
    info = determined.get_cluster_info()
    if info is not None and info.task_type == "TRIAL":
        # Managed trials.
        _context = core_v2._make_v2_context(
            ...
        )
        return _context

    # ok now we know we're not on cluster, we're doing unmanaged stuff now
    config = config or Config(
        project=defaults.project or unmanaged.project,
        workspace=defaults.workspace or unmanaged.workspace,
        ...
    )

    # now we can make the unmanaged context, this could be renamed to _init_unmanaged_context
    _context = _init_context(
        config=config or _merge_config(defaults, unmanaged),
        ...
    )
...
```
yeah, we'd have a chunk of duplicated code between the two init methods, but in general i feel like validation/deprecation checks should close to the user-facing method anyway.

i dunno. i'm nitpicking here, i think it's fine as is, just something i thought about. totally up to you.